### ISSUE TITLE
KFMon generator

### ISSUE THREAD

**Issue by geek1011** [geek1011/NickelMenu#29]
After #4 is finished and #28 has been merged.

**Comment by NiLuJe** [comment]
First stab ready at https://github.com/geek1011/NickelMenu/tree/niluje/kfmon-generator ;).

**Comment by geek1011** [comment]
Looks good upon a quick glance. Can you rebase it on master and open a PR?

**Comment by NiLuJe** [comment]
Might be a while, my MOBO is acting up and decided to stop enumerating the old SSD where my system lives ^^.

BIOS can't see it, EFI bootloaders can't see it, but if I boot a SystemRescueCD on a stick, Linux finds it just fine. o_O.

Going to try to check if a connector got unseated later today, which is a possibility, since it's an old M.2 stuck in a 2.5" adapter that doesn't fit properly anywhere in my chassis, so it's more or less "resting" on top of a HDD.... >_<".

**Comment by geek1011** [comment]
> Going to try to check if a connector got unseated later today, which is a possibility, since it's an old M.2 stuck in a 2.5" adapter that doesn't fit properly anywhere in my chassis, so it's more or less "resting" on top of a HDD.... >_<".

Mine's taped to the side of the case, since the screw holes don't fit for some reason... At least SSDs don't vibrate ;).

**Comment by NiLuJe** [comment]
I am indeed using rEFInd ;). (Initially because of a weird issue with GRUB since a BIOS update, where basically I lost monitor signal in the Linux fb console. Since I don't use a greeter, and I startx manually, doing that blind was fun :D. Was possibly half an nVidia quirk that might have been fixed when I had to update a VBIOS thingy (!) to support the latest DP revision when I swapped monitors. Hopefully moot now that I've switched to an AMD card. Aaaanyway, yeah, rEFInd is great, and trying its own USB stick was my first stop, but no dice there either ;)).

The first hint that something was wrong was when I was thrown straight into Windows' bootloader (from the new install on the new NVME M.2. Because of course my MOBO enumerates NVME drives *after* SATA drives, so my proper ESP is actually still on the old SATA M.2).

TL;DR: That was a fun end of the night. :D.

**Comment by NiLuJe** [comment]
Okay, I'm back!

Apparently, yeah, shitty adapter and/or cable means the SATA lead doesn't lock on the adapter's side, so, wonky shit ensued. (I'm calling it: the Linux kernel has telekinetic powers :D).

My lame-ass rack adapter surprisingly held up, on the other hand, so, yay, didn't need to resort to taping it up or playing with rubber bands ;p.

Plus, spring cleaning!

----

Working on the rebase now ;).

**Comment by geek1011** [comment]
@NiLuJe, one quick question: is the item order deterministic?

**Comment by geek1011** [comment]
Also, is there a way for users to control the item order?

**Comment by NiLuJe** [comment]
That behaves mostly like NM entries themselves: inode order ;).

**Comment by geek1011** [comment]
Ok, then I'll just leave it as is for now.
